parser-typechecker/src/Unison/Codebase.hs

```diff
@@ -160,31 +156,26 @@ typeLookupForDependencies codebase = foldM go mempty
 transitiveDependencies
   :: (Monad m, Var v)
   => CL.CodeLookup v m a
-  -> Set Reference
-  -> Reference
-  -> m (Set Reference)
-transitiveDependencies code seen0 r = if Set.member r seen0
+  -> Set Reference.Id
+  -> Reference.Id
+  -> m (Set Reference.Id)
+transitiveDependencies code seen0 rid = if Set.member rid seen0
   then pure seen0
   else
-    let seen = Set.insert r seen0
-    in
-      case r of
-        Reference.DerivedId id -> do
-          t <- CL.getTerm code id
-          case t of
-            Just t ->
-              foldM (transitiveDependencies code) seen (Term.dependencies t)
-            Nothing -> do
-              t <- CL.getTypeDeclaration code id
-              case t of
-                Nothing        -> pure seen
-                Just (Left ed) -> foldM (transitiveDependencies code)
-                                        seen
-                                        (DD.dependencies (DD.toDataDecl ed))
-                Just (Right dd) -> foldM (transitiveDependencies code)
-                                         seen
-                                         (DD.dependencies dd)
-        Reference.Builtin{} -> pure seen
+    let seen = Set.insert rid seen0
+        getIds = Set.mapMaybe Reference.toId
+    in CL.getTerm code rid >>= \case
+      Just t ->
+        foldM (transitiveDependencies code) seen (getIds $ Term.dependencies t)
+      Nothing -> 
+        CL.getTypeDeclaration code rid >>= \case 
+          Nothing        -> pure seen
+          Just (Left ed) -> foldM (transitiveDependencies code)
+                                  seen
+                                  (getIds $ DD.dependencies (DD.toDataDecl ed))
+          Just (Right dd) -> foldM (transitiveDependencies code)
+                                   seen
+                                   (getIds $ DD.dependencies dd)
```

Comment:
This is mostly whitespace diff, apart from the calls to `getIds`.  Could have folded those into new methods in the `Term` and `DataDeclaration` modules?